### DISCUSSION TITLE
Create `DestructiveAction` interface

### DISCUSSION THREAD

**Discussion by hivokas** [laravel/nova-issues#3951] [Q&A] [answered]
I like `ShouldQueue` interface which instructs Laravel to queue the job. It would be great to `implements DestructiveAction` instead of `extends DestructiveAction`.

Use case: I've created my own base action that will be used in some cases. Sometimes I want to make the action destructive. It would be clean with interface but not as clean without interface.

**Comment by crynobone** [comment, marked as answer]
You can just override `jsonSerialize()` method:

```php
/**
 * Prepare the action for JSON serialization.
 *
 * @return array<string, mixed>
 */
public function jsonSerialize(): array
{
     return array_merge(parent::jsonSerialize(), [
          'destructive' => true, // or $this instanceof CustomDescructiveActionInterface
     ]);
}
```

**Reply by hivokas** [reply]
Thanks!
